bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/TypeTags.java

```diff
@@ -85,6 +85,7 @@ public class TypeTags {
 
     public static final int PARAMETERIZED_TYPE_TAG = READONLY_TAG + 1;
     public static final int TYPE_REFERENCED_TYPE_TAG = PARAMETERIZED_TYPE_TAG + 1;
+    public static final int CLIENT_TAG = TYPE_REFERENCED_TYPE_TAG + 1;
```

Comment:
Do we really need a separate tag? Related to https://github.com/ballerina-platform/ballerina-lang/pull/37494#discussion_r955892840.

Reply:
Yes, for cases like https://github.com/ballerina-platform/ballerina-lang/blob/a326d859d7b74f3143307a476b234ee175712ca7/bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TypeChecker.java#L1697-L1700

We cannot use `OBJECT_TYPE_TAG` for clients.

Reply:
Why can't we check for object tag and the client flag?

Reply:
We don't have a client flag in `TypeFlags.java`

Reply:
We have it at and use it at compile-time, right? We can add use it at runtime and add it to the flags. Should be `io.ballerina.runtime.api.flags.SymbolFlags` btw.

Reply:
Updated with [1f3ba1f](https://github.com/ballerina-platform/ballerina-lang/pull/37494/commits/1f3ba1fa2403bfd99c0dd830d9003032a86ba40b)